// @peculiar/asn1-tsp/build/cjs/pki_failure_info.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PKIFailureInfo = exports.PKIFailureInfoFlags = void 0;
const asn1_schema_1 = require("@peculiar/asn1-schema");
var PKIFailureInfoFlags;
(function (PKIFailureInfoFlags) {
    PKIFailureInfoFlags[PKIFailureInfoFlags["badAlg"] = 1] = "badAlg";
    PKIFailureInfoFlags[PKIFailureInfoFlags["badRequest"] = 2] = "badRequest";
    PKIFailureInfoFlags[PKIFailureInfoFlags["badDataFormat"] = 16] = "badDataFormat";
    PKIFailureInfoFlags[PKIFailureInfoFlags["timeNotAvailable"] = 8192] = "timeNotAvailable";
    PKIFailureInfoFlags[PKIFailureInfoFlags["unacceptedPolicy"] = 16384] = "unacceptedPolicy";
    PKIFailureInfoFlags[PKIFailureInfoFlags["unacceptedExtension"] = 32768] = "unacceptedExtension";
    PKIFailureInfoFlags[PKIFailureInfoFlags["addInfoNotAvailable"] = 65536] = "addInfoNotAvailable";
    PKIFailureInfoFlags[PKIFailureInfoFlags["systemFailure"] = 16777216] = "systemFailure";
})(PKIFailureInfoFlags || (exports.PKIFailureInfoFlags = PKIFailureInfoFlags = {}));
class PKIFailureInfo extends asn1_schema_1.BitString {
    toJSON() {
        const flag = this.toNumber();
        const res = [];
        if (flag & PKIFailureInfoFlags.addInfoNotAvailable) {
            res.push("addInfoNotAvailable");
        }
        if (flag & PKIFailureInfoFlags.badAlg) {
            res.push("badAlg");
        }
        if (flag & PKIFailureInfoFlags.badDataFormat) {
            res.push("badDataFormat");
        }
        if (flag & PKIFailureInfoFlags.badRequest) {
            res.push("badRequest");
        }
        if (flag & PKIFailureInfoFlags.systemFailure) {
            res.push("systemFailure");
        }
        if (flag & PKIFailureInfoFlags.systemFailure) {
            res.push("systemFailure");
        }
        if (flag & PKIFailureInfoFlags.timeNotAvailable) {
            res.push("timeNotAvailable");
        }
        if (flag & PKIFailureInfoFlags.unacceptedExtension) {
            res.push("unacceptedExtension");
        }
        if (flag & PKIFailureInfoFlags.unacceptedPolicy) {
            res.push("unacceptedPolicy");
        }
        return res;
    }
    toString() {
        return `[${this.toJSON().join(", ")}]`;
    }
}
exports.PKIFailureInfo = PKIFailureInfo;
