// @peculiar/asn1-tsp/build/cjs/time_stamp_req.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeStampReq = exports.TimeStampReqVersion = void 0;
const tslib_1 = require("tslib");
const asn1_schema_1 = require("@peculiar/asn1-schema");
const asn1_x509_1 = require("@peculiar/asn1-x509");
const message_imprint_1 = require("./message_imprint");
var TimeStampReqVersion;
(function (TimeStampReqVersion) {
    TimeStampReqVersion[TimeStampReqVersion["v1"] = 1] = "v1";
})(TimeStampReqVersion || (exports.TimeStampReqVersion = TimeStampReqVersion = {}));
class TimeStampReq {
    version = TimeStampReqVersion.v1;
    messageImprint = new message_imprint_1.MessageImprint();
    reqPolicy;
    nonce;
    certReq = false;
    extensions;
    constructor(params = {}) {
        Object.assign(this, params);
    }
}
exports.TimeStampReq = TimeStampReq;
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], TimeStampReq.prototype, "version", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: message_imprint_1.MessageImprint })
], TimeStampReq.prototype, "messageImprint", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.ObjectIdentifier, optional: true,
    })
], TimeStampReq.prototype, "reqPolicy", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.Integer,
        converter: asn1_schema_1.AsnIntegerArrayBufferConverter,
        optional: true,
    })
], TimeStampReq.prototype, "nonce", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.Boolean, defaultValue: false,
    })
], TimeStampReq.prototype, "certReq", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_x509_1.Extensions, optional: true, context: 0, implicit: true,
    })
], TimeStampReq.prototype, "extensions", void 0);
