// @peculiar/asn1-tsp/build/cjs/tst_info.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TSTInfo = exports.TSTInfoVersion = void 0;
const tslib_1 = require("tslib");
const asn1_schema_1 = require("@peculiar/asn1-schema");
const asn1_x509_1 = require("@peculiar/asn1-x509");
const accuracy_1 = require("./accuracy");
const message_imprint_1 = require("./message_imprint");
var TSTInfoVersion;
(function (TSTInfoVersion) {
    TSTInfoVersion[TSTInfoVersion["v1"] = 1] = "v1";
})(TSTInfoVersion || (exports.TSTInfoVersion = TSTInfoVersion = {}));
class TSTInfo {
    version = TSTInfoVersion.v1;
    policy = "";
    messageImprint = new message_imprint_1.MessageImprint();
    serialNumber = new ArrayBuffer(0);
    genTime = new Date();
    accuracy;
    ordering = false;
    nonce;
    tsa;
    extensions;
    constructor(params = {}) {
        Object.assign(this, params);
    }
}
exports.TSTInfo = TSTInfo;
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.Integer })
], TSTInfo.prototype, "version", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.ObjectIdentifier })
], TSTInfo.prototype, "policy", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: message_imprint_1.MessageImprint })
], TSTInfo.prototype, "messageImprint", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.Integer, converter: asn1_schema_1.AsnIntegerArrayBufferConverter,
    })
], TSTInfo.prototype, "serialNumber", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({ type: asn1_schema_1.AsnPropTypes.GeneralizedTime })
], TSTInfo.prototype, "genTime", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: accuracy_1.Accuracy, optional: true,
    })
], TSTInfo.prototype, "accuracy", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.Boolean, defaultValue: false,
    })
], TSTInfo.prototype, "ordering", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_schema_1.AsnPropTypes.Integer,
        converter: asn1_schema_1.AsnIntegerArrayBufferConverter,
        optional: true,
    })
], TSTInfo.prototype, "nonce", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_x509_1.GeneralName, context: 0, optional: true,
    })
], TSTInfo.prototype, "tsa", void 0);
tslib_1.__decorate([
    (0, asn1_schema_1.AsnProp)({
        type: asn1_x509_1.Extensions, context: 1, implicit: true, optional: true,
    })
], TSTInfo.prototype, "extensions", void 0);
